src/hooks/safe-apps/useChainIdQueryParam.ts

```diff
@@ -0,0 +1,34 @@
+import { useRouter } from 'next/router'
+import useChains from '@/hooks/useChains'
+import { useMemo } from 'react'
+
+type ReturnType = {
+  chainId?: string
+  validChainId: boolean
+  loading: boolean
+  error: string | undefined
+}
+
+const useChainIdQueryParam = (): ReturnType => {
```

Comment:
Is there a reason we are not reusing the `chain` query param e.g. `?chain=eth`?

When I try to access a share url like `share/safe-app?appUrl=https://apps.gnosis-safe.io/drain-safe&chainId=1` it still display Rinkeby as the selected network.

Reply:
I wondered if it's possible to switch the network and somehow didn't find this param. Will adjust the share link now